doc/modules/ROOT/pages/index.adoc

```diff
@@ -0,0 +1,320 @@
+    "selfLink": "/apis/custom.metrics.k8s.io/v1beta1/namespaces/default/services/%2A/on_heap_ratio"
+  },
+  "items": [
+    {
+      "describedObject": {
+        "kind": "Service",
+        "namespace": "default",
+        "name": "hazelcast-metrics",
+        "apiVersion": "/v1"
+      },
+      "metricName": "on_heap_ratio",
+      "timestamp": "2020-11-13T12:57:23Z",
+      "value": "108m",
+      "selector": null
+    }
+  ]
+}
+----
+
+The most important part in this output is "value": `108m`. The suffix `m` means milli-unit, and it is https://github.com/DirectXMan12/k8s-prometheus-adapter/blob/master/docs/walkthrough.md#quantity-values[kubernetes-style quantities] to define metric values. Milli-unit is equivalent to 1000ths of a unit so `108m` is actually referring to 3.3% which means `max` value of `on_heap_ratio` seen so far.
+
+=== Install Horizontal Pod AutoScaler (Custom Metrics)
+
+As we have configured Hazelcast, Prometheus and Prometheus Adapter, let's create a Horizontal Pod AutoScaler based on `on_heap_ratio` metric. `hazelcast-custom-metrics-hpa.yaml` tells HPA if `targetValue > 200m` then scale up my cluster. `200m` as we explained above means actually 20%. You can change that number based on your own use case.
+
+See the content of hazelcast-custom-metrics-hpa.yaml file.
+
+[source, yaml]
+----
+apiVersion: autoscaling/v2beta2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: heap-based-hpa
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: StatefulSet
+    name: hazelcast
+  minReplicas: 3
+  maxReplicas: 10
+  metrics:
+    - type: Object
+      object:
+        describedObject:
+          kind: Service
+          name: hazelcast-metrics
+        metric:
+          name: on_heap_ratio
+        target:
+          type: Value
+          value: 200m
+----
+
+Apply HPA to your cluster with `kubectl`:
+
+[source, shell]
+----
+$ kubectl apply -f hazelcast-custom-metrics-hpa.yaml
+horizontalpodautoscaler.autoscaling/heap-based-hpa created
+----
+
+=== Generate some Memory Load for HPA
+
+Let's have a look `TARGETS` part of HPA output
+
+[source, shell]
+----
+$ kubectl get hpa heap-based-hpa
+NAME             REFERENCE               TARGETS     MINPODS   MAXPODS   REPLICAS   AGE
+heap-based-hpa   StatefulSet/hazelcast   180m/200m   3         10        10         11m
+----
+
+As we see, current HPA Target is 180m/200m so if we increase memory usage just 10%  by adding 10MB into the cluster, HPA should trigger a scale up event. I will use Hazelcast Java Client to put some data into cluster, but you can use your own language to implement the same functionality. You can see all Hazelcast supported programming languages https://hazelcast.org/imdg/clients-languages/[here].
+
+Let's first port forward from our local machine to be able to connect remote k8s hazelcast member pod.
+
+[source, shell]
+----
+$ kubectl port-forward hazelcast-0 5701
+----
+
+Execute following code snippet to put data into Hazelcast Cluster.
```

Comment:
the snippet is just a client instantiation. Let's either expand the snippet or indicate that this is the config needed to connect to the cluster and data should be inserted explicitly.